app/public/js/profile.js

```diff
@@ -106,12 +106,16 @@ function getCountryElement()
 
 function updateRegionOptions()
 {
-	var country_id = parseInt(getCountryElement().val());
-	var $datalist = $('#regions');
-	$datalist.empty();
+	let country_id = parseInt(getCountryElement().val());
+	let home_region = $('#home_region');
+	home_region.empty();
 	regions.forEach(function(region) {
 		if ( region.country_id === country_id )
-			$datalist.append($('<option />').text(region.name));
+			if (home_region.data('value') === region.name) {
+                home_region.append($('<option />').val(region.name).attr('selected', 'selected').text(region.name));
+			} else {
+                home_region.append($('<option value='+region.name+'/>').text(region.name));
```

Comment:
Use .val like you did a few lines up instead of writing value=... 

.val will save you from problems if the region.name happened to contain quotes or >.  I know that may be a weird edge case but it is pretty easy to make sure that's never a problem by using the val method.